vaadin-messages-flow-parent/vaadin-messages-flow/src/main/resources/META-INF/resources/frontend/messageListConnector.js

```diff
@@ -26,9 +26,9 @@ window.Vaadin.Flow.messageListConnector = {
       minute: 'numeric'
     });
     list.items = items.map((item) =>
-      item.time
+      item.__unformattedTime
         ? Object.assign(item, {
-            time: formatter.format(new Date(item.time))
+            time: formatter.format(new Date(item.__unformattedTime))
```

Comment:
An alternative would be to:
- extract a function for formatting the time on an array of items and call that in both `setItems` and `addItems`
- make `addItems` set the `items` property directly

I think that might be a bit simpler overall? No need to add a new property then, also less data sent to the client.

Another alternative would be to make the connector set a flag on already processed items and then just ignore those when mapping in ´setItems´.